packages/with_bloc/lib/src/with_bloc.dart

```diff
@@ -1,35 +1,53 @@
 import 'package:collection/collection.dart';
-import 'package:flutter/foundation.dart';
-import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
 
 const listId = ListEquality<dynamic>();
 
+/// A helper widget for easily creating [StateQueue] in the [build] method.
+///
+/// This widget takes care of creating the [BlocType] and disposing it when necessary.
+/// It also handles recreating the [BlocType] whenever any [inputs] change.
 class WithBloc<BlocType extends ValueNotifier<StateType>, StateType>
     extends StatefulWidget {
   const WithBloc({
     Key key,
     @required this.createBloc,
-    this.inputs,
     @required this.builder,
+    this.inputs = const [],
     this.child,
   })  : assert(createBloc != null),
         assert(builder != null),
+        assert(inputs != null),
         super(key: key);
 
+  /// A function which creates the [BlocType] instance
+  ///
+  /// This function might be called multiple times in the lifetime of this widget.
+  /// Whenever the [inputs] change, a new BLoC will be created by calling this function.
   final BlocType Function(BuildContext context) createBloc;
 
+  /// A [Function] which builds a widget depending on the [BlocType] and [StateType].
+  ///
+  /// The [bloc] will be the current bloc and [value] is the current [StateType].
+  /// When the bloc emits a new value, this function will be called again to reflect the changes in the UI.
+  ///
+  /// The [child] will be passed in as the fourth argument and might be `null` if not supplied.
   final Widget Function(
     BuildContext context,
     BlocType bloc,
     StateType value,
     Widget child,
   ) builder;
 
+  /// A [BlocType]-independent widget which is passed back to the [builder].
+  ///
+  /// This argument is optional and can be null if the entire widget subtree
```

Comment:
```
/// This argument is optional and can be `null`.
```
and what does `if the entire widget subtree the [builder] builds depends on the value of the [BlocType].` means ?

I couldn't follow the comment very well. do we pass the child as a performance optimisation ? so the child would not rebuild on each `[builder]` execution ?

Reply:
Yes, if you have a part of the subtree which is independent of the bloc and the value this can be a small optimization.

Reply:
https://github.com/aboutyou/dart_packages/pull/14/commits/f1b90ae220119eeea543dabf53e87140c3afda7e

Reply:
I am not sure if it is how you structure a sentence but with commas I can read it now.


also 
```suggestion
  ///
  /// The benefit of passing a widget down as a `child` is that it does 
  /// not get recreated on every `[builder]` execution. 
  ///
  /// This argument is optional and can be `null` if the entire widget subtree,
  /// which the [builder] builds, depends on the value of the [BlocType].

```

Reply:
doesn't matter with your new changes 👍